### ISSUE TITLE
gRPC-C++ cocoapod 1.40.0 seems broken

### ISSUE THREAD

**Issue by wu-hui** [grpc/grpc#27387]
### What version of gRPC and what language are you using?
gRPC-C++ cocoapod 1.40.0 

### What operating system (Linux, Windows,...) and version?
macOS

### What runtime / compiler are you using (e.g. python version or version of gcc)
Python 3.8.9
cc Apple clang version 12.0.5 (clang-1205.0.22.11)

### What did you do?
1. Create a xcode project `MyApp`.
2. Create `Podfile` as below
```
source 'https://github.com/CocoaPods/Specs.git'

platform :ios, '9.0'
inhibit_all_warnings!

target 'MyApp (iOS)' do
  pod 'gRPC-C++', '~> 1.40.0'

  # Has its own copy of OCMock
  # and has access to GoogleAnalytics via the app
  # that hosts the test target

  target 'Tests iOS' do
    inherit! :search_paths
    pod 'OCMock', '~> 2.0.1'
  end
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    puts target.name
  end
end
```
3. Run `pod install`
4. Open up the generated `xcworkspace` file from xcode, build the app. 

### What did you expect to see?
It builds.

### What did you see instead?

Observe errors like missing headers, for example `openssl/asn1.h`.

**Comment by wu-hui** [comment]
The issue starts to happen with `1.39.0` while `1.38.0` works, FYI.

**Comment by dennycd** [comment]
thanks for reporting this wu-hui, this looks like might be a mismatched boringssl version that is being used. Can you do a clean pod install (rm -Rf Pods Podfile.lock) and paste your installed pod dependency versions, it should looks sth like this 

```
Analyzing dependencies
Downloading dependencies
Installing BoringSSL-GRPC (0.0.19)
Installing abseil (1.20210324.0)
Installing gRPC-C++ (1.40.0)
Installing gRPC-Core (1.40.0)
Generating Pods project
BoringSSL-GRPC
abseil
gRPC-C++
gRPC-C++-gRPCCertificates-Cpp
gRPC-Core
Pods-MyApp
Integrating client project
```

**Comment by wu-hui** [comment]
Did a clean install with 1.39.0, problem remains.

```
➜  MyApp git:(main) ✗ rm -rf Pods
➜  MyApp git:(main) ✗ vim Podfile
➜  MyApp git:(main) ✗ rm Podfile.lock
➜  MyApp git:(main) ✗ pod install
Analyzing dependencies
Downloading dependencies
Installing BoringSSL-GRPC (0.0.19)
Installing OCMock (2.0.1)
Installing abseil (1.20210324.0)
Installing gRPC-C++ (1.39.0)
Installing gRPC-Core (1.39.0)
Generating Pods project
BoringSSL-GRPC
OCMock
abseil
gRPC-C++
gRPC-C++-gRPCCertificates-Cpp
gRPC-Core
Pods-MyApp (iOS)
Pods-Tests iOS
Integrating client project
Pod installation complete! There are 2 dependencies from the Podfile and 5 total pods installed.
```

**Comment by wu-hui** [comment]
Update:

doing
1. pod cache clean --all
2. rm -rf Pods/
3. rm Podfile.lock
4. pod install

Fixed the problem. Somehow some hooks failed to run when upgrade to 1.39.

**Comment by dennycd** [comment]
Linking to some prior issue related to this one 
https://github.com/grpc/grpc/issues/18886

tl;dr is that the pre-install hook in BoringSSL-GRPC pod failed for some reason on user's machine when upgrading to BoringSSL-GRPC 0.0.19. It rewrites header include path to point to <openssl_grpc/ instead of <opensll. in order to avoid conflict w/ openssl in system installs.   On script failure, all subsequent pod install would be using the cached version of. this BoringSSL.

**Comment by dennycd** [comment]
resolve this one with follow up internal cleanup in #27413
